subsys/net/ip/dhcpv4.c

```diff
@@ -880,14 +886,48 @@ static bool dhcpv4_parse_options(struct net_pkt *pkt,
 
 			break;
 		}
-		default:
-			NET_DBG("option unknown: %d", type);
+		default: {
+			struct net_dhcpv4_option_callback *cb, *tmp;
```

Comment:
The commit message says:
> Callbacks can only be registered for options that are not handled
internally (see `dhcpv4_parse_options` in dhcpv4.c for more info).

Which is a consequence of processing callbacks in the `default` label (unrecognized option). I wonder, is it really needed to introduce such a limitation? Unless I miss something, it should be possible to register callback for any option, if the option processing was moved before the "internal" processing (`switch` instruction). The net_pkt cursor is backed up, so it should be fairly transparent.

Reply:
Additional remark - i think such a functionality wouldn't be used that often, so it'd make sense to add a Kconfig option for it, just so that we don't increase the library footprint when not needed.

Reply:
Thanks for your review! I really like your suggestion of enabling this for any option. Even though it wouldn't be necessary for my use case, this might still be useful to others. I'll refactor it so that this is moved before the `switch` block.